File: src/doerly-fe/src/app/modules/payments/models/payment-history-item-response.ts
import {EPaymentStatus} from 'app/modules/payments/enums/e-payment-status';
import {ECurrency} from 'app/modules/payments/enums/e-currency';

export interface PaymentHistoryItemResponse {
  paymentId: number;
  createdAt: string;
  status: EPaymentStatus;
  amount: number;
  currency: ECurrency;
  description: string;
  billId: string;
  billDescription: string;
}

File: src/main-service/Doerly.Module.Payments/Doerly.Module.Payments.Contracts/Responses/PaymentHistoryItemResponse.cs
using Doerly.Module.Payments.Enums;

namespace Doerly.Module.Payments.Contracts;

public class PaymentHistoryItemResponse
{
    public int PaymentId { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; }
    public ECurrency Currency { get; set; }
    public EPaymentAction Action { get; set; }
    public EPaymentStatus Status { get; set; }
    public int BillId { get; set; }
    public DateTime CreatedAt { get; set; }
    
}
